### DISCUSSION TITLE
Editor

### DISCUSSION THREAD

**Discussion by Adrianotiger** [Adrianotiger/desktopPet#58] [General]
Any question about the editor

**Comment by cyborg90-cyber** [comment]
Where can we find sprites made by other people excluding the ones which  come with default?

**Reply by Adrianotiger** [reply]
Here you can find some resources:
https://adrianotiger.github.io/desktopPet/2019/07/02/sprite-sheets.html

Hope this can help you.

**Comment by stinkymichelle** [comment]
Hi! I really like esheep application! So nostalgic :) One question, could you make a tutorial on how to use editor program? I'm a graphic designer so I have a sprite sheet, but there's still some stuff I don't understand on creating the pet. My pet keeps walking backwards during testing! Nor does it "fall" down, even when I put an animation when detecting gravity. But, please take your time.

**Comment by Adrianotiger** [comment]
Hi Michelle,
I made a tutorial, but can't find the link anymore :P 

I have an old tutorial (from the online editor), you can find something there:
https://esheep.petrucci.ch/Help/Manual%20-%20online%20editor.pdf

The editor creates just an xml code, so all attributes of this animation can be read in the wiki:
https://github.com/Adrianotiger/desktopPet/wiki/XML%5CStructure

Why backwards? It is funny... Everyone will get this problem at beginning. It is because in Japan it is normal to read from right to left, so the pet always begins to walk from right to left ;)

To see if there is an error, I need the animation-xml. I am happy to help you when you need something! To test your animation, it is better if you use the original application and not the "simulator" inside the editor.

Once you have the xml, just drag this xml over the sheep and the new animation will be loaded. To test all your animations, you can close the original application and reopen it, pressing the SHIFT-key. This will open a debug window and you can open a hidden menu when you press over your pet with the right mouse button.

**Reply by stinkymichelle** [reply]
Okay, I'll try some of the stuff you said. I tested in the actual program by dragging and you're right, it is a lot better. I already found out why my pet walked backwards now haha. Still not sure why he doesn't fall, though! Well, I'll keep trying. I'm not a programmer so it's a bit difficult for me, haha. Thank you for replying so quickly.

**Comment by Adrianotiger** [comment]
![image](https://user-images.githubusercontent.com/7373079/168981346-a410a9f7-672a-47f4-8c91-7848ea956566.png)

Sequence:
sprite items 2 and 3 are played. Repeats [20] time.
So the animation will play 40 frames in this example.

Start / End parameters:
the first frame is played with the parameters from start, the last frame from the end parameters. frame 2 to 39 are interpolated (example start: -10, end -20. Frame 20 will be -15).
x, y: how much the pet should move on each played frame. -2 (will move your pet to the left). if 'y' is <0 then your pet will move down on the screen.
Interval: how much time from 1 frame to the other. Remember that this is a background application, so values under 30-50 will use much CPU. Try to use higher intervals so that the application will not use pc-resources.
Once all frames are played, the next animation will be played. If no animation is set, the pet will respawn.
If a border is detected, the next animation for the border is played. If no animation is set, the pet will walk outside your screen and will respawn.
If there is no window and no taskbar under the pet, the next animation for gravity is played. If there is no animation for gravity, the pet will continue to walk on the air.

You can write me a mail if you want, or we can continue skype or editor-chat.

**Reply by stinkymichelle** [reply]
Thank you for the information. I will let you know if I have more questions. How do I find your email?

**Reply by Adrianotiger** [reply]
oh yes :)
You can write me "adosapps" at outlook.

**Comment by GenesysJ** [comment]
Hello, I just discover the esheep and I just want to practice with animation, With your wonderfull editor offline, I create my xml (all your test success), really easy to use ♥.
But when I launch with the cmd `E:\>DesktopPet.exe localxml=animation.xml,` this just launch DesktopPet.exe without my animation. I don't understand how can I add my personal animation.

Thanks in advance

**Comment by Adrianotiger** [comment]
Hi Genesys,
thank you for the feedback and for trying the editor :D
Yes, it should work with your command.
If it is just for testing it, you can drop your xml file on the active pet.
If there is something wrong or if the xml fails to load, the default pet will be loaded. Probably, this is what happens.

If it is just to try, it is better if you drop the xml directly on your pet. You can start also your application, pressing SHIFT when it starts. So you will get a debug window and can see if the application is throwing some errors.

**Reply by Adrianotiger** [reply]
Hi. It isn't a problem with the editor but with the application. I updated the application and should now be able to load local and web animations. If you use "animations.xml", it search on your exe path. If you have it located somewhere in D:, you need to add the full path.

**Reply by kissko** [reply]
_A comment that was deleted by mistake, to which Adrianotiger responded above:_

Hi Adrianotiger.
Thanks for your app. It's awesome.

I have the same problem.

I downloaded release 1.3.1.0, and the animations.xml files from the "Pets" section.
For example, blue_sheep\animations.xml

I tried running the command D:\>DesktopPet.exe localxml=animations.xml , but the default pet loaded.
I tried dropping animations.xml on DesktopPet.exe with the mouse. The default pet loaded again.
I tried doing this on Windows 10 on a real PC and on Windows 11 in VirtualBox. No result, the default pet.
I tryed start also application, pressing SHIFT when it starts. The log file is no different from running without animations.xml.

I tried to do "install application", but on both systems the shortcuts for launching were inoperative, it simply does not launch. If you find the installed EXE, then it launches. By the way, during installation the downloaded online pet is lost, and although the icon of the installed application was from blue_sheep, the default pet launched.

The only working way to launch blue_sheep without Internet access (online pet) turned out to be this. I took blue_sheep\animations.xml , converted it to one line. Then converted some characters (< `&lt;` " `&quot;` > `&gt;`) and copied it to the DesktopPet file in the line `<add key="xml" value=" `. Then finally blue_sheep pet started. But this is a very non-trivial way.

**How to run DesktopPet with custom animations in a normal way?**

PS: In my opinion, the best way would be to have DesktopPet automatically load animations from the animations.xml file if it is found in the EXE directory.

**Reply by kissko** [reply]
> I updated the application and should now be able to load local and web animations.

Thank you very much. In this release 1.3.2, running custom animations from the command line with localxml works well.

PS dropping the xml by mouse directly on pet working too. At first I thought that I needed to drop it on the EXE file, but in fact I needed to drop it on the sheep (pet) itself on the screen.

**Comment by GenesysJ** [comment]
Thank you so much, well with the command line this doesn't work, I think I miss something, but when I drop the xml on the pet it work

**Comment by JesusGod-Pope666-Info** [comment]
Who made those upgraded 7 Sheep, is that person still around?
Maybe he could also make some video's how he does it and put them on a video platform for others to more quickly get it.
Could be cool to have even more animations for those 7 sheep :)

Hmm, I am thinking, there must be some animation schools out there in the world - where, they could do some more animations for the Sheep as a project. That would be cool. Anyone have any ideas, then we could send out some emails to them and see if they would bite. Like think about a whole animation class taking it up, maybe in groups for producing each one animation more for the sheep. Like that would quickly then give us even more to enjoy! :P

**Comment by JesusGod-Pope666-Info** [comment]
I think I might have another idea as well to spread it more around so people get to know about this project. And extract more people who wanna work on it.

**Comment by Edward1Stark** [comment]
Thank god you are not like one of those accounts who can fix a problem but their last message is from 6 years ago 😂 hello dear how are you doing? Sorry i know this is not related to the topic but I need your help with my problem with emulating tags with pn532 and esp32, as far as I've seen, you've had so many successful attempts, would you mind helping me for my project?

And for being in touch choose whatever you like:
Email: edward1stark@gmail.com
Telegram: @edward1stark_main
Or even here in this topic

Just for you know, I'm using seed studio but there is something wrong, I've tried so much I don't even know what the problem is but I can explain all the things I've tried etc...
